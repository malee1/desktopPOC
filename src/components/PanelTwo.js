import { React, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Box, Typography, List, ListItem, Button, Collapse } from '@material-ui/core';
import { AccessTime, RecordVoiceOver, Help, Mic, ArrowDropDown } from '@material-ui/icons';
import { useSelector } from 'react-redux';
import { panelTwoLabels } from '../constants/panelLabels';
import DropDownPanel from './DropDownPanel';
import DropDownPanel2 from './DropDownPanel2';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    padding: '5px'
  },
  box: {
    width: '27vw'
  },
  list: {
    flexDirection: 'column',
    whiteSpace: 'nowrap'
  },
  listItem: {
    backgroundColor: 'white'
  },
  listItemButton: {
    justifyContent: 'flex-end',
    backgroundColor: 'white',
    display: 'flex'
  },
  iconContainer: {
    width: '8%',
    backgroundColor: 'white'
  },
  icon: {
    backgroundColor: 'white'
  },
  labelContainer: {
    width: '40%',
    marginLeft: '2px'
  },
  label: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  labelValueContainer: {
    width: '55%',
    marginLeft: '2px'
  },
  labelValue: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  dropDown: {
    // backgroundColor: 'grey',
    zIndex: '1000',
    position: 'absolute',
    height: '15vh',
    width: '27vw'
  },
  panelButton: {
    height: '30px'
  }
});

export default function PanelOne() {
  const classes = useStyles();

  const labels = panelTwoLabels;

  const customer = useSelector((state) => state.customer.data);
  const user = useSelector((state) => state.user.data.roleProfile);

  const [open, setOpen] = useState(false);

  const handleDropDown = () => {
    setOpen(!open);
  };

  let dropDown;
  if (user === 'Gamma') {
    dropDown = <DropDownPanel2 />;
  } else {
    dropDown = <DropDownPanel />;
  }

  return (
    <div className={classes.root} data-testid="panel-two">
      <Box className={classes.box} bgcolor="primary.main">
        <List className={classes.list}>
          <ListItem className={classes.listItem} data-testid="item">
            <div className={classes.iconContainer}>
              <AccessTime className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.time}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.timeWait}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem} data-testid="item">
            <div className={classes.iconContainer}>
              <Help className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.context}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.context}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem} data-testid="item">
            <div className={classes.iconContainer}>
              <Mic className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.rec}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.recording}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem} data-testid="item">
            <div className={classes.iconContainer}>
              <RecordVoiceOver className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.info}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.callInfo}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItemButton}>
            <Button
              className={classes.panelButton}
              onClick={handleDropDown}
              variant="contained"
              color="secondary"
            >
              More Context
              <ArrowDropDown />
            </Button>
          </ListItem>
        </List>
        <Collapse in={open} timeout="auto" unmountOnExit>
          <div className={classes.dropDown}>{dropDown}</div>
        </Collapse>
      </Box>
    </div>
  );
}
