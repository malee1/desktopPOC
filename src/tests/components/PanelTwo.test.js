import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

import PanelTwo from '../../components/PanelTwo';

const mockStore = configureMockStore();
const store = mockStore({
  customer: {
    data: {
      name: 'Test Test',
      dob: '1111',
      uRef: '2222',
      tel: '3333',
      timeWait: 'aaaa',
      context: 'bbbb',
      recording: 'cccc',
      callInfo: 'dddd',
      alerts: 'Important Info 1',
      service: 'Alert1',
      keyInfo: 'Info 1',
      auth: 'Approved'
    }
  },
  user: {
    data: {
      fileId: '123456'
    }
  }
});

const store2 = mockStore({
  customer: {
    data: {
      name: 'Test Test',
      dob: '1111',
      uRef: '2222',
      tel: '3333',
      timeWait: 'aaaa',
      context: 'bbbb',
      recording: 'cccc',
      callInfo: 'dddd',
      alerts: 'Important Info 1',
      service: 'Alert1',
      keyInfo: 'Info 1',
      auth: 'Approved'
    }
  },
  user: {
    data: {
      fileId: '234567',
      roleProfile: 'Gamma',
      boltOn: 'none'
    }
  }
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('the panel renders', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PanelTwo />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has one component', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <PanelTwo />
      </Provider>
    );
    const container = getAllByTestId('panel-two');
    expect(container.length).toBe(1);
  });

  it('has four items', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <PanelTwo />
      </Provider>
    );
    const container = getAllByTestId('item');
    expect(container.length).toBe(4);
  });
});

describe('it shows the correct content', () => {
  it('has the correct name', () => {
    const { getAllByText } = render(
      <Provider store={store}>
        <PanelTwo />
      </Provider>
    );
    // const container = getAllByText('Test Test');
    expect(getAllByText('aaaa').length).toBe(1);
    expect(getAllByText('bbbb').length).toBe(1);
    expect(getAllByText('cccc').length).toBe(1);
    expect(getAllByText('dddd').length).toBe(1);
  });
});

describe('button click renders correct drop panel', () => {
  it('shows Drop Panel 1 on button click', () => {
    const { getByText, getAllByTestId } = render(
      <Provider store={store}>
        <PanelTwo />
      </Provider>
    );
    const btn = getByText('More Context');
    fireEvent.click(btn);
    const dropPanel = getAllByTestId('drop-down-panel');
    expect(dropPanel.length).toBe(1);
  });
});

describe('button click renders correct drop panel', () => {
  it('shows Drop Panel 2 on button click', () => {
    const { getByText, getAllByTestId } = render(
      <Provider store={store2}>
        <PanelTwo />
      </Provider>
    );
    const btn = getByText('More Context');
    fireEvent.click(btn);
    const dropPanel = getAllByTestId('drop-down-panel-2');
    expect(dropPanel.length).toBe(1);
  });
});
