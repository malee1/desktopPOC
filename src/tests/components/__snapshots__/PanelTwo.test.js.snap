// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the panel renders matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="panel-two"
>
  <div
    className="MuiBox-root MuiBox-root-14 makeStyles-box-2"
  >
    <ul
      className="MuiList-root makeStyles-list-3 MuiList-padding"
    >
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Time Waiting: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            aaaa
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Context: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            bbbb
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Recording: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            cccc
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="9"
              cy="9"
              r="4"
            />
            <path
              d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm7.76-9.64l-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27zM20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Call Info: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            dddd
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItemButton-5 MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-panelButton-13 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            More Context
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
